include/SFML/System/Vector2.inl

```diff
@@ -53,6 +53,138 @@ y(static_cast<T>(vector.y))
 }
 
 
+////////////////////////////////////////////////////////////
+template <typename T>
+T Vector2<T>::length() const
+{
+    static_assert(std::is_floating_point_v<T>, "Vector2::length() is only supported for floating point types");
```

Comment:
The message here is not very helpful, it is redundant with the type trait predicate itself, which does appear as part of compiler diagnostics. I suggest explaining *why* instead:

```suggestion
    static_assert(std::is_floating_point_v<T>, "Calculating the length of an integer vector is a lossy operation");
```

Or something like that.

Reply:
It's not redundant, the default error message is one of those that make C++ (in)famous, at least on VS 2019. I only have a German one installed on this machine, but it basically says "when compiling the class template member function", i.e context only.
```cpp
1>{path}/SFML/System/Vector2.inl(60,24): error C2338: Vector2::length() is only supported for floating point types
1>{path}/SFML/System/Vector2.inl(59): message : beim Kompilieren der Klasse Vorlage-Memberfunktion "T sf::Vector2<T>::length(void) const"
1>        with
1>        [
1>            T=int
1>        ]
1>{path}/examples\tennis\Tennis.cpp(33): message : Siehe Verweis auf die gerade kompilierte Instanziierung "T sf::Vector2<T>::length(void) const" der Funktions-Vorlage.
1>        with
1>        [
1>            T=int
1>        ]
1>{path}/SFML/Graphics/Image.hpp(217): message : Siehe Verweis auf die gerade kompilierte Klasse Vorlage-Instanziierung "sf::Vector2<int>".
1>Die Erstellung des Projekts "tennis.vcxproj" ist abgeschlossen -- FEHLER.
========== Erstellen: 0 erfolgreich, 1 fehlerhaft, 5 aktuell, 0 übersprungen ==========
```

I think the "why" part makes sense to mention, but maybe rather in the documentation? The reason is the same for all the methods: they cannot be meaningfully/correctly implemented for integers.